webapp/test_results_handler.go

```diff
@@ -21,7 +22,12 @@ import (
 //
 // The browsers initially displayed to the user are defined in browsers.json.
 // The JSON property "initially_loaded" is what controls this.
-func testHandler(w http.ResponseWriter, r *http.Request) {
+func testResultsHandler(w http.ResponseWriter, r *http.Request) {
+	if strings.Index(r.URL.Path, "/results/") != 0 {
+		http.Redirect(w, r, fmt.Sprintf("/results%s", r.URL.Path), 308)
```

Comment:
Never seen 308 Permanent Redirect before, but if it's cached like 301 Moved Permanently and unlike 302 Found that could be a problem. (https://whatwg.org/faq was once a 301 and in many people's caches, broke things a bit when we wanted to host the FAQ there.) We might want to put something on / later, right?

Reply:
> similar to 301 (Moved Permanently) but does not allows the request method to be changed from POST to GET .

Changed to 307.